dataprofiler/profilers/profile_builder.py

```diff
@@ -2112,6 +2130,161 @@ def _update_chi2(self):
 
         return chi2_mat
 
+    def _get_null_and_sample_count(self):
+        """
+        Gets the null_count and sample_size of each column.
+
+        :return: Dictionary where column id maps to a list of [null_count, sample_size]
+        :rtype: dict
+        """
+        counts = {}
+        for col_id in range(len(self._profile)):
+            profile = self._profile[col_id]
+            null_count = getattr(profile, "null_count", 0)
+            sample_size = getattr(profile, "sample_size", 0)
+            counts[col_id] = [null_count, sample_size]
+        return counts
+
+    def _calc_null_replication_metrics(self, clean_samples, prev_counts):
+        """
+        Calculate metrics needed for replicating null values in capitalone/synthetic-data.
+
+        Metrics include class priors and class specific means of each column.
+
+        Required for running LDA based binary classifier where predicted class label indicates whether
+        a value of a column should be NaN (1) or not (0).
+
+        :param clean_samples: input cleaned dataset
+        :type clean_samples: dict
+        :param prev_counts: null_count and sample_size of each column before update
+        :type prev_counts: dict
+        """
+        data = pd.DataFrame(clean_samples).apply(pd.to_numeric, errors="coerce")
+
+        for col_id in range(len(self._profile)):
+            profile = self._profile[col_id]
+            null_count = getattr(profile, "null_count")
+            if null_count == 0:
+                # No missing values to replicate
+                continue
+
+            # Calculate class priors
+            # i.e probability that a value in target col is or is not NaN
+            sample_size = getattr(profile, "sample_size")
+            true_count = sample_size - null_count
+
+            # null_count and sample_size get updated on profile_update
+            # Therefore the priors are calculated from the updated values
+            prior_null = null_count / sample_size
+            prior_not_null = true_count / sample_size
+
+            # Partition data based on whether target column value is null or not
+            col_is_null = data.iloc[:, col_id].isnull()
+            partition_null = data[col_is_null]
+            partition_not_null = data[~col_is_null]
+
+            # Drop the target column from partitions
+            partition_null = partition_null.drop(columns=col_id)
+            partition_not_null = partition_not_null.drop(columns=col_id)
+
+            # Calculate mean vectors of each partition
+            # i.e mean vectors of rows where target col is or is not null
+            mean_not_null, mean_null = (
+                partition_not_null.mean().tolist(),
+                partition_null.mean().tolist(),
```

Comment:
what if we were to calc this relative to the mean we know for each column against union of null to non-null in the other columns?

Reply:
Also, do we need to calc the mean of both? We should know the mean of the entire column, and the mean of one partial, hence we could calculate the other without O(n)